recipes/sdl/all/patches/0003-2.0.14-wayland-scanner-buildrequires.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/cmake/sdlchecks.cmake b/cmake/sdlchecks.cmake
+index a9434c8..67f809e 100644
+--- a/cmake/sdlchecks.cmake
++++ b/cmake/sdlchecks.cmake
+@@ -593,20 +593,11 @@ endmacro()
+ # - HAVE_DLOPEN opt
+ macro(CheckWayland)
+   if(VIDEO_WAYLAND)
+-    pkg_check_modules(WAYLAND wayland-client wayland-scanner wayland-egl wayland-cursor egl xkbcommon)
++    pkg_check_modules(WAYLAND wayland-client wayland-egl wayland-cursor egl xkbcommon)
+ 
+     if(WAYLAND_FOUND)
+-      execute_process(
+-        COMMAND ${PKG_CONFIG_EXECUTABLE} --variable=wayland_scanner wayland-scanner
```

Comment:
I think we should patch wayland recipe to make it populate the wayland_scanner variabel in thewayland-scanner.pc. This patch would not be needed

Reply:
It is not enough. A `wayland-scanner.pc` will be generated from the package in the _host_ context and we would need the one from the _build_ context... I'm not sure if `.pc` files accept _build modules_ and we can do something similar to the one in `protobuf` to find the proper executable.

IMO, CMake takes the tools it uses to build from the environment. So it does with the compiler or the linker and I like to have the same approach for any other tool used during the build process. In my mind, all these things are basically the same: tools used by CMake, and it is very convenient to expect all of them to provide the same kind of information (or to be available via the same mechanism).

Reply:
ok, so when build_requiring and requiring the same package, it is known that the environment variables of the build requirement take precedence over the requirement ?

Reply:
(Using two-profiles) Only the environment variables from the _build_ context are propagated to the `build()` method (no precedence to worry about).

Reply:
@jgsogo 
Is there discussion about ways to combine info from a recipe used as a build requirement + host requirement?

Reply:
IMHO, I think it's something that needs to be decided on the Conan client side first.

AFAIK, **CMake** doesn't have the concept of build-requires, so generating `FindXXX.cmake` or `XXXX-config.cmake` for them would be misleading, as there is no `find_package(XXXX HOST/BUILD_CONTEXT)` and a way to distinguish the targets coming from different contexts. In **Meson** I think there are contexts and you can find dependencies from each of them and Meson knows how to handle them (I've just read, I didn't try it).

Maybe we should generate all the files for all the packages in both graphs. Meson will know how to use them, CMake will require work on the consumer side to decide which one to use and find (meanwhile, CMake's assumption of build tools being available in the environment is IMO the best we can do).